custom/enikshay/reports/case_finding.py

```diff
@@ -0,0 +1,131 @@
+# coding=utf-8
+
+from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumn
+from custom.enikshay.reports.consts import AGE_RANGES
+from custom.enikshay.reports.generic import EnikshayReport, EnikshayMultiReport
+from custom.enikshay.reports.sqldata.case_finding_sql_data import CaseFindingSqlData
+from custom.enikshay.reports.treatment_outcome import AllTBPatientsReport
+
+
+def get_for_all_ranges(slug, data):
+    row = []
+    for lower_bound, upper_bound in AGE_RANGES[:-1]:
+        row.append(data.get('%s_age_%d_%d' % (slug, lower_bound, upper_bound), 0))
+
+    row.append(data.get('%s_age_%d' % (slug, AGE_RANGES[-1][0]), 0))
+    row.append(data.get('%s_total' % slug, 0))
+    return row
+
+
+class CaseFindingAllTBPatientsReport(AllTBPatientsReport):
+    report_title = 'Block 1: All TB patients registered in the quarter'
+
+    @property
+    def rows(self):
+        return super(CaseFindingAllTBPatientsReport, self).rows[1:4]
+
+
+class AllNewAndRecurrentTBCases(EnikshayReport):
+    name = 'Block 2: All new and recurrent TB cases only: from column above'
+    slug = 'all_new_and_recurrent_tb_cases'
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn('Age'),
+            DataTablesColumn('0-4'),
+            DataTablesColumn('5-14'),
+            DataTablesColumn('15-24'),
+            DataTablesColumn('25-34'),
+            DataTablesColumn('35-44'),
+            DataTablesColumn('45-54'),
+            DataTablesColumn('55-64'),
+            DataTablesColumn(u'≥65'),
```

Comment:
nit: it might be nice if these columns were automatically generated from `AGE_RANGES`